packages/@expo/cli/src/start/server/DevelopmentSession.ts

```diff
@@ -81,10 +82,14 @@ export class DevelopmentSession {
 
   /** Try to close any pending development sessions, but always resolve */
   public async closeAsync(): Promise<boolean> {
-    if (env.CI || env.EXPO_OFFLINE) {
+    if (env.CI || env.EXPO_OFFLINE || !this.hasActiveSession) {
       return false;
     }
 
+    // Clear out the development session, even if the call fails.
+    // This blocks subsequent calls to `stopAsync`
+    this.hasActiveSession = false;
```

Comment:
should only be set to false after a successful invocation to the endpoint

Reply:
As discussed, we want to avoid calling `/notify-close` when server is down. This does exactly that, clearing it out to only let it pass once. (And then again when the session was restarted)